src/c_cpp_properties.cc

```diff
@@ -0,0 +1,136 @@
+#include "c_cpp_properties.h"
+
+#include <fstream>
+#include <iostream>
+#include <regex>
+
+#include <rapidjson/document.h>
+#include <doctest/doctest.h>
+
+#ifdef _WIN32
+  #define CURRENT_PLATFORM "Win32"
+#elif defined(__APPLE__)
+  #define CURRENT_PLATFORM "Mac"
+#else
+  #define CURRENT_PLATFORM "Linux"
+#endif
+
+static std::string current_platform = CURRENT_PLATFORM;
+
+static
+optional<CCppProperties> LoadCCppPropertiesFromStr(
+    const char* filecontent,
+    const std::string& project_dir) {
+  CCppProperties res;
+  res.args.push_back("%clang");
+  rapidjson::Document document;
+  document.Parse(filecontent);
+  if (!document.IsObject())
+    return {};
+  auto conf_it = document.FindMember("configurations");
+  if (conf_it == document.MemberEnd())
+    return {};
+  if (!conf_it->value.IsArray())
+    return {};
+  for (auto& conf : conf_it->value.GetArray()) {
+    if (!conf.HasMember("name") || conf["name"].GetString() != ::current_platform)
```

Comment:
after kCurrentPlatform name change I believe you can remove the `::` global scope access